### PR TITLE
[Hub Generated] Review request for Microsoft.Batch to add version stable/2022-10-01

### PR DESCRIPTION
**Pull request by wiboris**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/batch-Microsoft.Batch-2022-10-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

- Deperated Certificates
-- Marked all Certificate related api's as Deprecated
-- Updated descriptions of Certificate related apis to indicate that the api's will be deprecated by Feb 2024
- Add api support for node communication mode setting
-- Added definition for NodeCommunicationMode supporting types default, classic, and simplified
-- Added targetNodeCommunicationMode and currentNodeCommunicationMode property to PoolProperties definition

- S360 changes
-- Fixed a CASE issue with properties "**dynamicVNetAssignmentScope**" of the NetworkConfiguration definition.  Changed to dynamicVnetAssignmentScope by making the 'N' lowercase.  The api has always been returning lowercase 'n' but the SDK have been able to handle it.
-- Fixed a CASE issue with properties "**username**" of the CIFSMountConfiguration definition.  Changed to userName by making the 'N' uppercase.  The api has always been returning uppercase 'N' but the SDK have been able to handle it.
-- Renamed broken properties of the PrivateLinkServiceConnectionState definition from to **actionRequired** to **actionsRequired**.  The properties was incorrect in the swagger and was a missing 's' and thus this parameter never gave a value.
--set "x-nullable": true attribute of allowedAuthenticationModes property of BatchAccountProperties definition
--added "x-ms-mutability": ["read", "update"]": attribute of data property of CertificateCreateOrUpdateProperties definition
--added "x-ms-mutability": ["create", "update"]": attribute of nodeDeallocationOption property of FixedScaleSettings definition

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 1]
```diff
@@ -3943,7 +3947,7 @@
           "type": "string",
           "title": "Description of the private Connection state"
         },
-        "actionRequired": {
+        "actionsRequired": {
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:3950]: This is to fix a S360 issue that was detected.  Our api was returning actionsRequired and not actionRequired.  This was broken for a while

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 2]
```diff
@@ -5444,7 +5452,7 @@
             "description": "Setting up RBAC for Azure Batch VNets"
           }
         },
-        "dynamicVNetAssignmentScope": {
+        "dynamicVnetAssignmentScope": {
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:5455]: This is to fix a S360 issue that was detected. Our api was returning dynamicVnetAssignmentsScope and not dynamicVNetAssignmentsScope. This was broken for a while

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 3]
```diff
@@ -6093,7 +6101,7 @@
     "CIFSMountConfiguration": {
       "type": "object",
       "properties": {
-        "username": {
+        "userName": {
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:6104]: This is to fix a S360 issue that was detected. Our api was returning userName and not username. This was broken for a while

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 4]
```diff
@@ -6117,7 +6125,6 @@
         }
       },
       "required": [
-        "username",
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: 360 issue. The username filed was not always returned

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 5]
```diff
@@ -6154,7 +6161,6 @@
       "required": [
         "accountName",
         "azureFileUrl",
-        "accountKey",
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: s360 issue, the accountKey was not always returned

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 6]
```diff
@@ -2993,7 +2993,7 @@
         },
         "allowedAuthenticationModes": {
           "readOnly": true,
-          "x-nullable": false,
+          "x-nullable": true,
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:2996]: 360 ROUNDTRIP_MISSING_PROPERTY issue with Get response not having this value

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 7]
```diff
@@ -3656,7 +3656,11 @@
         "data": {
           "type": "string",
           "title": "The base64-encoded contents of the certificate.",
-          "description": "The maximum size is 10KB."
+          "description": "The maximum size is 10KB.",
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:3659]: 360 ROUNDTRIP_MISSING_PROPERTY was not present in the Get response

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json [hunk 8]
```diff
@@ -4357,7 +4361,11 @@
         "nodeDeallocationOption": {
           "title": "Determines what to do with a node and its running task(s) if the pool size is decreasing.",
           "description": "If omitted, the default value is Requeue.",
-          "$ref": "#/definitions/ComputeNodeDeallocationOption"
+          "$ref": "#/definitions/ComputeNodeDeallocationOption",
```
Comment (wiboris) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json:4364]: s360 issue ROUNDTRIP_MISSING_PROPERTY not returned in the get method

**Comment by zizw123** [comment]
This  PR is labeled with breaking change, if that's the case, please get it approved first and come back to ARM review. Thanks.

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 2 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 4 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 5 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 6 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 7 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

File: specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json (hunk 8 above)
Comment (wiboris) [reply specification/batch/resource-manager/Microsoft.Batch/stable/2022-10-01/BatchManagement.json]: reverted

**Comment by leni-msft** [comment]
@tadelesh please help review the Go SDK breaking changes

**Comment by tadelesh** [comment]
Approved for Go breakings as removal of unreferenced types are acceptable.

**Review by leniatgh — APPROVED** [review]

**Pull request merged**